syncstorage/views/__init__.py

```diff
@@ -295,6 +315,9 @@ def post_collection(request):
     bsos = request.validated["bsos"]
     invalid_bsos = request.validated["invalid_bsos"]
 
+    if request.validated["batch"] or request.validated["commit"]:
+        return post_collection_transaction(request)
+
```

Comment:
It may be worth pulling the original non-transactional implementation out into a separate `post_collection_single` function or similar, to make it more obvious that we're choosing between two independent implementations.